src/patterns/gen-ai/aws-summarization-appsync-stepfn/index.ts

```diff
@@ -705,8 +705,9 @@ export class SummarizationAppsyncStepfn extends Construct {
     this.processedAssetBucket?.grantReadWrite(documentReaderLambda);
 
 
-    const enableOperationalMetric = props.enableOperationalMetric || true;
-    const solution_id = 'genai_cdk_'+id;
+    const enableOperationalMetric =
+      props.enableOperationalMetric !== undefined && props.enableOperationalMetric !== null ? props.enableOperationalMetric : true;
+    const solution_id = 'genai_cdk_' + version + '/' + typeof(this) + '/' + id;
 
     if (enableOperationalMetric) {
       documentReaderLambda.addEnvironment(
```

Comment:
consider moving solutionId instantiation behind the conditional if solutionId isn't used elsewhere